sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -138,6 +138,80 @@ object JdbcUtils extends Logging {
       throw new IllegalArgumentException(s"Can't get JDBC type for ${dt.simpleString}"))
   }
 
+  // A `JDBCValueGetter` is responsible for converting, getting a value from `Row`
+  // and then setting into a field for `PreparedStatement`. The last argument
+  // `Int` means the index for the value to be set in the SQL statement and also used
+  // for the value to retrieve from `Row`.
+  private type JDBCValueGetter = (PreparedStatement, Row, Int) => Unit
```

Comment:
hmm, I think `setter` is a more proper name here, and we should use `getter` for the read path.

`JDBCValueGetter`, sounds like we get the value from JDBC.